### PR TITLE
Replace CLI-based configuration with file-based (#133)

### PR DESCRIPTION
**Pull request by paulgb**
This replaces the previous CLI-based configuration with a layered configuration:
- If a positional argument is passed to the drone on startup, configuration is loaded from there.
- If environment variables matching `SPAWNER_*` are present, they override the configuration.

Additionall, a `--dump-config` flag can be used to dump the configuration as a JSON blob instead of running the drone. This is useful for validation and testing of env overrides.

The only required argument is `cluster_domain`, so this works:

    SPAWNER_CLUSTER_DOMAIN=spawner.test cargo run --bin spawner-drone -- --dump-config

However, to do anything useful, at least one of the optional `[acme]`, `[proxy]`, or `[agent]` sections must be provided.

A more complete example is provided in the `sample_config.toml` file in root. It can be validated with this command:

    cargo run --bin spawner-drone -- sample_config.toml --dump-config

### CODE REVIEW

File: sample_config.toml [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# This file contains some examples of Drone configuration.
+# You can run the drone with --dump-config to validate a configuration.
```
Comment (pretentious7) [inline sample_config.toml:2]: this is not implemented yet?

**Review by pretentious7 — APPROVED** [review]
lgtm!

ah this is going to make life a lot nicer.

File: sample_config.toml (hunk 1 above)
Comment (paulgb) [reply sample_config.toml:2]: It's here: https://github.com/drifting-in-space/spawner/blob/a3dc709adb38451dfb73fb9a8c419aee868fefea/drone/src/drone/mod.rs#L50

Comment (pretentious7) [reply sample_config.toml:2]: ah my bad i keep forgetting the - to _ thing while grepping

**Pull request merged**